src/utilities/strategy-helpers/blue-green-helper.ts

```diff
@@ -0,0 +1,329 @@
+    let shouldBeRouted: boolean = false;
+    const serviceSelector: string = getServiceSelector(serviceObject);
+    if (!!serviceSelector) {
+        deploymentEntityList.every((depObject) => {
+            // finding if there is a deployment in the given manifests the service targets
+            const matchLabels: string = getDeploymentMatchLabels(depObject); 
+            if (!!matchLabels && isServiceSelectorSubsetOfMatchLabel(serviceSelector, matchLabels)) {
+                shouldBeRouted = true;
+                return false;
+            }
+        });
+    }
+    return shouldBeRouted;
+}
+
+export function createWorkloadsWithLabel(kubectl: Kubectl, deploymentObjectList: any[], nextLabel: string) {
+    const newObjectsList = [];
+    deploymentObjectList.forEach((inputObject) => {
+        // creating deployment with label
+        const newBlueGreenObject = getNewBlueGreenObject(inputObject, nextLabel);
+        core.debug('New blue-green object is: ' + JSON.stringify(newBlueGreenObject));
+        newObjectsList.push(newBlueGreenObject);
+    });
+    const manifestFiles = fileHelper.writeObjectsToFile(newObjectsList);
+    const result = kubectl.apply(manifestFiles);
+
+    return { 'result': result, 'newFilePaths': manifestFiles };
+}
+
+export function getNewBlueGreenObject(inputObject: any, labelValue: string): object {
+    const newObject = JSON.parse(JSON.stringify(inputObject));
+
+    // Updating name only if label is green label is given
+    if (labelValue === GREEN_LABEL_VALUE) {
+        newObject.metadata.name = getBlueGreenResourceName(inputObject.metadata.name, GREEN_SUFFIX);
+    }
+
+    // Adding labels and annotations
+    addBlueGreenLabelsAndAnnotations(newObject, labelValue);
+
+    return newObject;
+}
+
+export function addBlueGreenLabelsAndAnnotations(inputObject: any, labelValue: string) {
+    //creating the k8s.deploy.color label
+    const newLabels = new Map<string, string>();
+    newLabels[BLUE_GREEN_VERSION_LABEL] = labelValue;
+
+    // updating object labels and selector labels
+    helper.updateObjectLabels(inputObject, newLabels, false);
+    helper.updateSelectorLabels(inputObject, newLabels, false);
+
+    // updating spec labels if it is a service
+    if (!helper.isServiceEntity(inputObject.kind)) {
+        helper.updateSpecLabels(inputObject, newLabels, false);
+    }
+}
+
+export function getBlueGreenResourceName(name: string, suffix: string) {
+    return `${name}${suffix}`;
+}
+
+export function getSpecLabel(inputObject: any): string {
+    if(!!inputObject && inputObject.spec && inputObject.spec.selector && inputObject.spec.selector.matchLabels && inputObject.spec.selector.matchLabels[BLUE_GREEN_VERSION_LABEL]) {
+        return inputObject.spec.selector.matchLabels[BLUE_GREEN_VERSION_LABEL]; 
+    }
+    return '';
+}
+
+export function getDeploymentMatchLabels(deploymentObject: any): string {
```

Comment:
Instead of returning `string`, return `any`.
Return the `matchLabels` object itself. Return null in place of `''`. Use it as a dictionary in the subsequent code.